File: 프로그래머스/1/159994. 카드 뭉치/카드 뭉치.swift
import Foundation

func solution(_ cards1:[String], _ cards2:[String], _ goal:[String]) -> String {
    var cardArr1 = cards1
    var cardArr2 = cards2
    var goalArr = goal
    
    while !goalArr.isEmpty {
        let word = goalArr.removeFirst()
        
        if let first = cardArr1.first, first == word {
            cardArr1.removeFirst()
        } else if let first = cardArr2.first, first == word {
            cardArr2.removeFirst()
        } else {
            return "No"
        }
    }
    
    return "Yes"
}

File: 프로그래머스/1/159994. 카드 뭉치/카드 뭉치.cpp
#include <algorithm>
#include <iostream>
#include <queue>
#include <string>
#include <vector>

using namespace std;

string solution(vector<string> cards1, vector<string> cards2, vector<string> goal) {
    queue<string> q1, q2;
    int cnt = 0;

    for (auto str : cards1) {
        q1.push(str);
    }

    for (auto str : cards2) {
        q2.push(str);
    }

    for (int i = 0; i < goal.size(); i++) {
        if (!q1.empty() && q1.front() == goal[i]) {
            q1.pop();
            cnt++;
        }
        if (!q2.empty() && q2.front() == goal[i]) {
            q2.pop();
            cnt++;
        }
    }

    return (cnt == goal.size()) ? "Yes" : "No";
}